inst/doc/pitfalls.Rmd

```diff
@@ -0,0 +1,51 @@
+---
+title: "Common Pitfalls and Solutions"
+author: "Alison Appling"
+date: "February 4, 2019"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Shared cache}
+  %\VignetteEngine{knitr::rmarkdown}
+  \usepackage[utf8]{inputenc}
+---
+
+<head>
+<style>
+sym {color: red;}
+sol {color: blue;}
+</style>
+</head>
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, cache=FALSE, collapse=TRUE)
+```
+
+The following bullets describe problems or frustrations that `scipiper` users might encounter, along with some ideas for how to interpret and fix those issues. Each "pitfall" is presented in this format: _<sym>Problem or symptom.</sym> Explanation. <sol>Solution.</sol>_
+
+## General use of scipiper/remake
+
+* <sym>Unnecessary re-runs of functions.</sym> `scipiper` and `remake` don't update their databases when you run a function outside of scipiper, even if the function creates the desired outputs. This may be frustrating if you've just run a long-running function outside of `scipiper` and are now attempting to use `scipiper` to build targets that depend on the function's output, because `scipiper` will insist on running that function again. <sol>Anytime you might want the output of a function to be considered complete, build the `scipiper` target rather than calling the function directly.</sol>
+
+* <sym>Common debugging methods don't work.</sym> If you build a `scipiper` target that runs function `A()` to create target `a`, setting RStudio checkpoints or `debug(A)` won't let you debug function `A` when you call `scmake('a')`. <sol>Temporarily insert a call to `browser()` within the function instead.</sol>
+
+* <sym>The GitHub repo is too big.</sym> You might be committing data files or large task remake files to the repo. <sol>In general, you can safely .gitignore all data files, the .remake/ folder, and (optionally) any task remake files.</sol>
+
+## Use of shared cache
+
+* <sym>Teammates aren't getting updates about available intermediate data files.</sym> When sharing intermediate data files with a team using a shared cache, the status information about those files is contained in (1) a \*.ind file promising the existence of the data file in the cache, and (2) a build/status/\*.yml file telling `remake` how to decide whether the \*.ind file is up to date. <sol>Always `git commit` the \*.ind and build/status/\*.yml files corresponding to all shared data files.</sol>
+
+* <sym>Teammates can't see updated objects.</sym> R objects can't be shared. <sol>To share objects, save them as .rds files and use the standard shared cache syntax.</sol>
```

Comment:
is this the preferred format? (.rds)

Reply:
Well...I prefer it to .RData, and besides .RData it's the other format that's great at handling any R object type you throw at it, but I didn't mean to restrict it here; .feather, .tsv, and anything else would be just fine, too. I'll change the language.